### ISSUE TITLE
Negative dimension size caused by subtracting 5 from 1 for 'conv1d_63/convolution/Conv2D'

### ISSUE THREAD

**Issue by alejandrods** [keras-team/keras#9244]
Hi!

I am trying to train my CNN for text classification. I am following this code: 

https://github.com/keras-team/keras/blob/master/examples/pretrained_word_embeddings.py

But I have a problem because when I'm training I get this error:

_Negative dimension size caused by subtracting 5 from 1 for 'conv1d_63/convolution/Conv2D' (op: 'Conv2D') with input shapes: [?,1,1,128], [1,5,128,128]._

My code is this:

    data = data[indices]
    labels = labels[indices]
    num_validation_samples = int(VALIDATION_SPLIT * data.shape[0])
    
    x_train = data[:-num_validation_samples]
    y_train = labels[:-num_validation_samples]
    x_val = data[-num_validation_samples:]
    y_val = labels[-num_validation_samples:] 
    
    EMBEDDING_DIM = 100         #Si ponemos 100 cambia el acc
    MAX_NUM_WORDS = 20000
    
    GLOVE_DIR = "./data/"
    embeddings_index = {}
    f = open(os.path.join(GLOVE_DIR, 'glove.6B.100d.txt'))
    for line in f:
        values = line.split()
        word = values[0]
        coefs = np.asarray(values[1:], dtype='float32')
        embeddings_index[word] = coefs
    f.close()
    
    empty = []
    i=0
    embedding_matrix = np.random.random((len(word_index) + 1, EMBEDDING_DIM))
    for word, i in word_index.items():
        embedding_vector = embeddings_index.get(word)
        if embedding_vector is not None:
            # words not found in embedding index will be all-zeros.
            embedding_matrix[i] = embedding_vector
            print('Empty')
            empty.append(i)
            i += 1

    embedding_layer = Embedding(len(word_index) + 1,
                            EMBEDDING_DIM,
                            weights=[embedding_matrix],
                            input_length=MAX_SEQUENCE_LENGTH,
                            trainable=True)

    sequence_input = Input(shape=(MAX_SEQUENCE_LENGTH,), dtype='int32')
    embedded_sequences = embedding_layer(sequence_input)
    l_cov1= Conv1D(128, 5, activation='relu')(embedded_sequences)
    l_pool1 = MaxPooling1D(5)(l_cov1)
    l_cov2 = Conv1D(128, 5, activation='relu')(l_pool1)
    l_pool2 = MaxPooling1D(5)(l_cov2)
    l_cov3 = Conv1D(128, 5, activation='relu')(l_pool2)
    l_pool3 = GlobalMaxPooling1D()(l_cov3)  # global max pooling
    l_flat = Flatten()(l_pool3)
    l_dense = Dense(128, activation='relu')(l_flat)
    preds = Dense(len(labels_index), activation='softmax')(l_dense)
    
    model = Model(sequence_input, preds)
    model.compile(loss='categorical_crossentropy',
                  optimizer='rmsprop',
                  metrics=['acc'])
    
    model.summary()
    model.fit(x_train, y_train, validation_data=(x_val, y_val),
          nb_epoch=10, batch_size=128)

And my data:
 **x_train = (20846, 10), y_train = (20846, 2198), x_val = (5211, 10), y_val = (5211, 2198)**

My len(labels_index) = 2198

I don't know where is the problem because I have followed the code.

**Comment by anj-s** [comment]
Can you print out the input data and label shapes?
